src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs

```diff
@@ -153,6 +153,13 @@ private bool TryGetTextForSymbol(SyntaxToken token, SemanticModel semanticModel,
                 symbol = ((ILocalSymbol)symbol).Type;
             }
 
+            // Range variable: use the type
+            if (symbol is IRangeVariableSymbol)
+            {
+                var info = semanticModel.GetSpeculativeTypeInfo(token.SpanStart, SyntaxFactory.ParseName(symbol.Name), SpeculativeBindingOption.BindAsExpression);
+                symbol = info.Type;
```

Comment:
@gafter That sounds complicated :)   Given that, it definitely seems like something we'd want a compiler API for, rather than just parsing and trying to speculatively bind names.

> rather each occurrence of a range variable in a query expression may have a different type

Sounds like a good API would be one that took a range variable, and an occurrence location, and provided a type there.  Would that work?